docs/src/explanations/Documentation-of-functions.md

```diff
@@ -0,0 +1,641 @@
+---
+
+layout: docs
+
+title:  "Documention of functions"
+
+section: "chisel3"
+
+---
+
+# Documentation of Functions := vs <>
+
+
+
+The difference in the function of := and <> has been a major source of debate in lots of conversations.
+
+
+
+This document explains the difference by setting up an experiment using Scastie examples + DecoupledIO examples to put the functions to the test.
+
+
+
+### Experiment
+
+
+
+```scala mdoc:silent
+
+// Imports used by the following examples
+
+import chisel3._
+
+import chisel3.util.DecoupledIO
+
+```
+
+
+
+The diagram for the experiment can be viewed **[here](https://docs.google.com/document/d/14C918Hdahk2xOGSJJBT-ZVqAx99_hg3JQIq-vaaifQU/edit?usp=sharing)**
+
+
+
+
+
+```scala mdoc:silent
+
+class Wrapper extends Module{
+
+  val io = IO(new Bundle {
+
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+
+  val out = DecoupledIO(UInt(8.W))
+
+  })
+
+  val p = Module(new PipelineStage)
+
+  val c = Module(new PipelineStage) 
+
+  // connect Producer to IO
+
+  p.io.a <> io.in
+
+  // connect producer to consumer
+
+  c.io.a <> p.io.b
+
+  // connect consumer to IO
+
+  io.out <> c.io.b
+
+}
+
+
+
+class PipelineStage extends Module{
+
+  val io = IO(new Bundle{
+
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+
+    val b = DecoupledIO(UInt(8.W))
+
+  })
+
+  io.b <> io.a
+
+}
+
+```
+
+Below we can see the resulting Verilog for this example:
+
+```scala modc
+
+ChiselStage.emitVerilog(new Wrapper)
+
+```
+
+
+
+## Concept 1: <> is Communicative
+
+
+
+This experiment is set up to test for the function of "<>" using the experiment above.
+
+Achieving this involves flipping the RHS and LHS of each arrow and seeing how "<>"  will react.
+
+( Scastie link for the ecperiment:https://scastie.scala-lang.org/Shorla/LVhlbkFQQnq7X3trHfgZZQ )
+
+
+
+```scala mdoc:silent
```

Comment:
does this work? I imagine you might need to have `scala mdoc:silent:reset` otherwise you would get an error that Wrapper was already defined?